x-pack/plugins/observability_solution/infra/server/routes/metadata/lib/get_metric_metadata.ts

```diff
@@ -70,6 +72,20 @@ export const getMetricMetadata = async (
             size: 1000,
           },
         },
+        monitoredHost: {
+          filter: getFilterByIntegration(SYSTEM_INTEGRATION),
+          aggs: {
+            name: {
+              terms: {
+                field: HOST_NAME_FIELD,
+                size: 1,
+                order: {
+                  _key: 'asc',
+                },
+              },
+            },
+          },
+        },
```

Comment:
This endpoint is used for other asset types. We might want to run this agg only if `nodeType === 'host'`.

Also, perhaps running a query like this `getHasDataFromSystemIntegration` parallel to this query might be faster than this term agg because it early terminates as soon as it finds a document that satisfies the criteria.

Reply:
Hey! Thanks for adding the condition.

The fact that the query filters by `host.name` **helps**, but elasticsearch still needs to collect documents to build the aggregation. `getHasDataFromSystemIntegration` requires less computational - this function could even be reused. 

Another potential downside is if we need to do something similar for other asset types. I'm not sure if adding aggregations to this query will scale nicely.

But it's up to you. We can always revisit this in the future.

Reply:
Got it, thanks. Yeah, when I think about it it is a good idea - it brings dependency to the `infraMetricsClient` but I guess that is fine as we can use it in case we have similar queries for different assets.